tictactoe.py

```diff
@@ -0,0 +1,202 @@
+import math
+import random
+ 
+ 
+class TicTacToe:
+    """
+    a class for the TicTacToe game
+    """
+    def __init__(self, board_size = 3, win_length = 3, ai = True):
+        """
+        None for empty
+        True for O
+        False for X
+        """
+        self.board_size = board_size
```

Comment:
Assert that board_size has to be greater than 0 because we divide by this quantity later on.